lib/rules/function-component-definition.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * @fileoverview Standardize the way function component get defined
+ * @author Stefan Wullems
+ */
+
+'use strict';
+
+const Components = require('../util/Components');
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const NAMED_FUNCTION_TEMPLATES = {
+  'function-declaration': 'function $name$typeParams($params) $body',
+  'arrow-function': 'const $name$typeAnnotation =$typeParams($params) => $body',
+  'function-expression': 'const $name$typeAnnotation = function$typeParams($params) $body'
```

Comment:
tbh these should probably be `var`, so it's maximally compatible but the autofixes for `prefer-const` and `no-var` can kick in?